pypeit/images/scienceimage.py

```diff
@@ -0,0 +1,307 @@
+
+        Returns:
+            np.ndarray: Copy of the read noise squared image
+
+        """
+        msgs.info("Generating read noise image from detector properties and amplifier layout)")
+        # Convenience
+        detector = self.spectrograph.detector[self.det-1]
+        datasec_img = self.spectrograph.get_datasec_img(filename=self.filename,
+                                                        det=self.det)
+        # Build it
+        self.rn2img = procimg.rn_frame(datasec_img,
+                                       detector['gain'],
+                                       detector['ronoise'],
+                                       numamplifiers=detector['numamplifiers'])
+        # Return
+        return self.rn2img.copy()
+
+    def process_raw(self, filename, bias, pixel_flat, illum_flat=None):
+        """
+        Process a raw science image
+
+        Args:
+            filename (str):
+            bias (np.ndarray or None):
+            pixel_flat (np.ndarray):
+            illum_flat (np.ndarray, optional):
+
+        Returns:
+            np.ndarray: Copy of self.image
+
+        """
+        # Build up
+        prawImage = processrawimage.ProcessRawImage(filename, self.spectrograph,
+                                                    self.det, self.par,
+                                                    frametype=self.frametype)
+        # Save a bit
+        self.filename = filename
+        self.exptime = prawImage.exptime
+        # Process steps
+        process_steps = procimg.init_process_steps(bias, self.par)
+        process_steps += ['trim', 'apply_gain', 'orient']
+        if (pixel_flat is not None) or (illum_flat is not None):
+            process_steps += ['flatten']
+        # Do it
+        self.image = prawImage.process(process_steps, pixel_flat=pixel_flat,
+                                       bias=bias, illum_flat=illum_flat,
+                                       bpm=self.bpm, debug=True)
+        return self.image.copy()
+
+    def update_mask_cr(self, subtract_img=None):
+        """
+        Updates the CR mask values in self.mask
+        through a call to ImageMask.build_crmask which
+        generates a new CR mask and then a call to
+        ImageMask.update_mask_cr() which updates self.mask
+
+        Args:
+            subtract_img (np.ndarray, optional):
+                If provided, this is subtracted from self.image prior to
+                CR masking
+        """
+        # Generate the CR mask (and save in self.crmask)
+        _ = super(ScienceImage, self).build_crmask(self.spectrograph, self.det,
+                                                      self.par, self.image,
+                                                      utils.calc_ivar(self.ivar),
+                                                      subtract_img=subtract_img).copy()
+        # Now update the mask
+        _ = super(ScienceImage, self).update_mask_cr(self.crmask)
+
+    def __sub__(self, other):
```

Comment:
Is this __sub__ being used anywhere? I could not find any.

Reply:
this function was 'vetted'